### PR TITLE
Reduce service exit

### PR DESCRIPTION
**Pull request by thiagoftsm**
##### Summary
This PR reduces the following events :

```sh
8)
MAIN : SERVICE CONTROL: the following 1 service(s) [ COLLECTORS ] take too long to exit: 'PD[ebpf]' (97538); giving up on them...
```

on user hosts. With these changes eBPF will close faster, because new default options won't be expensive to host and it is checking more frequently status to close.

##### Test Plan

1. Compile branch
2. start netdata and wait few minutes.
3. Stop netdata and take a look in `error.log`.

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

**Review by MrZammler — DISMISSED** [review]
Tested on Ubuntu 22.04, seems to work fine!

File: collectors/ebpf.plugin/ebpf.c [hunk 1]
```diff
@@ -483,7 +483,17 @@ static void ebpf_exit()
...
         error("Cannot remove PID file %s", filename);
 
-    ebpf_exit_plugin = 2;
+#ifdef NETDATA_INTERNAL_CHECKS
+    error("Good bye world! I was PID %d", main_thread_id);
+#endif
+    printf("DISABLE\n");
+
+    if (!shm_ebpf_cgroup.header) {
+        munmap(shm_ebpf_cgroup.header, shm_ebpf_cgroup.header->body_length);
```
Comment (stelfrag) [inline collectors/ebpf.plugin/ebpf.c:492]: Shouldn't this be the opposite? If you do have a valid `shm_ebpf_cgroup.header` to unmap it?

Comment (thiagoftsm) [reply collectors/ebpf.plugin/ebpf.c]: You're right, I inverted the calls, thank you!

Comment (stelfrag) [reply collectors/ebpf.plugin/ebpf.c]: I meant the `if (!shm_ebpf_cgroup.header)` statement :smiley:  With this check, you will try to munmap a null pointer

Comment (thiagoftsm) [reply collectors/ebpf.plugin/ebpf.c:492]: Thanks, I fixed.

**Comment by stelfrag** [comment]
I can't pinpoint exactly how/when but sometimes it still takes long to complete eg.

grep on a specific PID

```
2023-02-02 20:09:04: netdata INFO  : PD[ebpf] : (0194@libnetdata/threads/t:thread_start   ): thread created with task id 126132
2023-02-02 20:09:04: netdata INFO  : PD[ebpf] : (0156@libnetdata/threads/t:thread_set_name): set name of thread 126132 to PD[ebpf]
2023-02-02 20:09:22: netdata INFO  : MAIN : (0275@daemon/main.c       :service_wait_ex): SERVICE CONTROL: waiting for the following 5 services [ COLLECTORS ] to exit: 'STATS_GLOBAL' (126103), 'STATS_WORKERS' (126104), 'P[diskspace]' (126116), 'P[proc]' (126118), 'PD[ebpf]' (126132)
2023-02-02 20:09:23: netdata INFO  : MAIN : (0275@daemon/main.c       :service_wait_ex): SERVICE CONTROL: waiting for the following 4 services [ COLLECTORS ] to exit: 'STATS_WORKERS' (126104), 'P[diskspace]' (126116), 'P[proc]' (126118), 'PD[ebpf]' (126132)
2023-02-02 20:09:23: netdata INFO  : MAIN : (0275@daemon/main.c       :service_wait_ex): SERVICE CONTROL: waiting for the following 2 services [ COLLECTORS ] to exit: 'P[diskspace]' (126116), 'PD[ebpf]' (126132)
2023-02-02 20:09:23: netdata INFO  : MAIN : (0275@daemon/main.c       :service_wait_ex): SERVICE CONTROL: waiting for the following 1 services [ COLLECTORS ] to exit: 'PD[ebpf]' (126132)
2023-02-02 20:09:24: netdata INFO  : MAIN : (0275@daemon/main.c       :service_wait_ex): SERVICE CONTROL: waiting for the following 1 services [ COLLECTORS ] to exit: 'PD[ebpf]' (126132)
2023-02-02 20:09:25: netdata INFO  : MAIN : (0275@daemon/main.c       :service_wait_ex): SERVICE CONTROL: waiting for the following 1 services [ COLLECTORS ] to exit: 'PD[ebpf]' (126132)
2023-02-02 20:09:26: netdata INFO  : MAIN : (0275@daemon/main.c       :service_wait_ex): SERVICE CONTROL: waiting for the following 1 services [ COLLECTORS ] to exit: 'PD[ebpf]' (126132)
2023-02-02 20:09:26: netdata INFO  : MAIN : (0290@daemon/main.c       :service_wait_ex): SERVICE CONTROL: the following 1 service(s) [ COLLECTORS ] take too long to exit: 'PD[ebpf]' (126132); giving up on them...
2023-02-02 20:09:26: netdata INFO  : MAIN : (0275@daemon/main.c       :service_wait_ex): SERVICE CONTROL: waiting for the following 2 services [ COLLECTORS ANALYTICS ] to exit: 'PD[ebpf]' (126132), 'ANALYTICS' (126358)
2023-02-02 20:09:26: netdata INFO  : MAIN : (0275@daemon/main.c       :service_wait_ex): SERVICE CONTROL: waiting for the following 1 services [ COLLECTORS ] to exit: 'PD[ebpf]' (126132)
2023-02-02 20:09:27: netdata INFO  : MAIN : (0275@daemon/main.c       :service_wait_ex): SERVICE CONTROL: waiting for the following 1 services [ COLLECTORS ] to exit: 'PD[ebpf]' (126132)
2023-02-02 20:09:28: netdata INFO  : MAIN : (0275@daemon/main.c       :service_wait_ex): SERVICE CONTROL: waiting for the following 1 services [ COLLECTORS ] to exit: 'PD[ebpf]' (126132)
2023-02-02 20:09:29: netdata INFO  : MAIN : (0275@daemon/main.c       :service_wait_ex): SERVICE CONTROL: waiting for the following 1 services [ COLLECTORS ] to exit: 'PD[ebpf]' (126132)
2023-02-02 20:09:30: netdata INFO  : MAIN : (0275@daemon/main.c       :service_wait_ex): SERVICE CONTROL: waiting for the following 1 services [ COLLECTORS ] to exit: 'PD[ebpf]' (126132)
2023-02-02 20:09:31: netdata INFO  : MAIN : (0275@daemon/main.c       :service_wait_ex): SERVICE CONTROL: waiting for the following 1 services [ COLLECTORS ] to exit: 'PD[ebpf]' (126132)
2023-02-02 20:09:32: netdata INFO  : MAIN : (0275@daemon/main.c       :service_wait_ex): SERVICE CONTROL: waiting for the following 1 services [ COLLECTORS ] to exit: 'PD[ebpf]' (126132)
2023-02-02 20:09:33: netdata INFO  : MAIN : (0275@daemon/main.c       :service_wait_ex): SERVICE CONTROL: waiting for the following 1 services [ COLLECTORS ] to exit: 'PD[ebpf]' (126132)
2023-02-02 20:09:34: netdata INFO  : MAIN : (0275@daemon/main.c       :service_wait_ex): SERVICE CONTROL: waiting for the following 1 services [ COLLECTORS ] to exit: 'PD[ebpf]' (126132)
2023-02-02 20:09:35: netdata INFO  : MAIN : (0275@daemon/main.c       :service_wait_ex): SERVICE CONTROL: waiting for the following 1 services [ COLLECTORS ] to exit: 'PD[ebpf]' (126132)
2023-02-02 20:09:36: netdata INFO  : MAIN : (0275@daemon/main.c       :service_wait_ex): SERVICE CONTROL: waiting for the following 1 services [ COLLECTORS ] to exit: 'PD[ebpf]' (126132)
2023-02-02 20:09:36: netdata INFO  : MAIN : (0290@daemon/main.c       :service_wait_ex): SERVICE CONTROL: the following 1 service(s) [ COLLECTORS ] take too long to exit: 'PD[ebpf]' (126132); giving up on them.
```

**Review by Dim-P — APPROVED** [review]
Started and stopped this branch multiple times on `Fedora 37` and `Ubuntu 22.04`, I saw the message in question only once. eBPF charts also look fine. LGTM!

**Comment by thiagoftsm** [comment]
> I can't pinpoint exactly how/when but sometimes it still takes long to complete eg.

If you take a look in `collector.log` and search for `Good bye world!` you will see that the collector called `exit()` seconds before last messages, so this difference between a call to `exit` and thread ends has relationship with OS return information for thread and of course do cleanup in some FD. 

This is the reason  said for you last Tuesday that I am planning to move from `callocs` and `mallocs` to our new array allocator,  I want to simplify OS' life, but I have understand that we are monitoring different functions inside kernel and when they are heavily used, it is necessary time to remove trampolines/probes/tracepoints.

**Pull request merged**